core/chains/evm/client/simulated_backend_client.go

```diff
@@ -64,37 +73,82 @@ func (c *SimulatedBackendClient) checkEthCallArgs(
 			"must be an eth.CallArgs, got %+#v", args[0])
 	}
 	blockNumber, err := c.blockNumber(args[1])
-	if err != nil || blockNumber.Cmp(c.currentBlockNumber()) != 0 {
+	if err != nil {
 		return nil, nil, fmt.Errorf("fourth arg to SimulatedBackendClient.Call "+
-			"must be the string \"latest\", or a *big.Int equal to current "+
-			"blocknumber, got %#+v", args[1])
+			"must be the string \"latest\", or a *big.Int, got %#+v", args[1])
+	}
+
+	toAddr, err := interfaceToAddress(callArgs["to"])
+	if err != nil {
+		return nil, nil, err
 	}
+
+	// to and from need to map to a common.Address but could come in as a string
 	ca := CallArgs{
-		From: callArgs["from"].(common.Address),
-		To:   *callArgs["to"].(*common.Address),
+		To:   toAddr,
 		Data: callArgs["data"].(hexutil.Bytes),
 	}
+
+	// from is optional in the standard client; default to 0x when missing
+	if value, ok := callArgs["from"]; ok {
+		addr, err := interfaceToAddress(value)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		ca.From = addr
+	} else {
+		ca.From = common.HexToAddress("0x")
+	}
+
 	return &ca, blockNumber, nil
 }
 
+func interfaceToAddress(value interface{}) (common.Address, error) {
+	switch v := value.(type) {
+	case common.Address:
+		return v
+	case string:
+		return common.HexToAddress(v), nil
+	case *big.Int:
+		return common.BigToAddress(v), nil
+	default:
+		return common.HexToAddress("0x"), fmt.Errorf("unrecognized value type for converting value to common.Address; try string, *big.Int, or common.Address")
+	}
+}
+
 // CallContext mocks the ethereum client RPC calls used by chainlink, copying the
 // return value into result.
+// The simulated backend cannot access old blocks and will return an error if
```

Comment:
Looks like the 1st paragraph here needs to be deleted?
The next para seems the right description as per this change